pkg/commands/options/filestuff.go

```diff
@@ -34,6 +34,24 @@ func AddFileArg(cmd *cobra.Command, fo *FilenameOptions) {
 		"Filename, directory, or URL to files to use to create the resource")
 	cmd.Flags().BoolVarP(&fo.Recursive, "recursive", "R", fo.Recursive,
 		"Process the directory used in -f, --filename recursively. Useful when you want to manage related manifests organized within the same directory.")
+
+	err := cmd.MarkFlagFilename("filename", "yaml", "yml", "json")
```

Comment:
Have you checked that the combination of `MarkFlagFilename` and `MarkFlagDirname` work together, to prefer shell completion for file names _or_ directory names? It's not clear from the docs that these won't conflict. I can also check if that would be helpful.

Reply:
sorry for coming back late as I was busy.

> Have you checked that the combination of `MarkFlagFilename` and `MarkFlagDirname` work together, to prefer shell completion for file names _or_ directory names? It's not clear from the docs that these won't conflict. I can also check if that would be helpful.

I can't get kubectl to work on my local machine, and the docs are not clear. but however I checked cobra's source code, and it handles this part by a map called annotations and filename has a key different from FlagDirname and it iterates on the keys while doing auto-completion.

so if you got some time can you check it's behavior?